spec/support/shared_examples/document_capture_concern.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+shared_examples_for 'DocumentCaptureConcern' do |next_step:|
```

Comment:
So for me, I would go with a more unit-test-like approach to testing this mixin. I think it would probably be worth doing a single happy-path check in the controllers for the headers to make sure it's all wired together correctly, but I wouldn't bother with a shared example (I find shared examples difficult to locate and harder to understand).


RSpec lets us make [anonymous controllers](https://relishapp.com/rspec/rspec-rails/docs/controller-specs/anonymous-controller) so we could make a test something like... (this is an outline, I haven't run this):

```ruby
# spec/controllers/concerns/idv/document_capture_concern_spec.rb
RSpec.describe Idv::DocumentCaptureConcern, type: :controller do
  controller do
    include Idv::DocumentCaptureConcern

    before_action :override_document_capture_step_csp

    def index; end
  end

  describe '#override_document_capture_step_csp' do
    it 'sets the headers for the document capture step' do
      get :index, params: { step: 'document_capture' }
      expect(...).to have_headers....
    end

   it 'does not set headers for any other step' do
      get :index, params: { step: 'some_other_step' }
      expect(...).to_not have_headers....
    end
  end
end
```

Reply:
Sounds good. The thinking behind the shared examples was to ensure that we're using the mixin where we're expecting to be using it, without having to duplicate a bunch of code, or to build-in knowledge of the mixin's implementation to the controller specs. To your point, I suppose this is more along the unhappy path of testing.